themes-default/slim/views/vue-components/name-pattern.mako

```diff
@@ -0,0 +1,491 @@
+        props: {
+            /**
+             * Current naming pattern.
+            */
+            namingPattern: {
+                type: String,
+                default: ''
+            },
+            /**
+             * An array of available preset naming patterns.
+            */
+            namingPresets: {
+                type: Array,
+                default: () => []
+            },
+            /**
+             * The selected multi ep style
+             */
+            multiEpStyle: {
+                type: Number,
+                default: null
+            },
+            /**
+             * Availale multi ep style
+             */
+            multiEpStyles: {
+                type: Array,
+                default: () => []
+            },
+            /**
+             * For anime shows there are a number of variations on how the absolute episode number
+             * is added to the episode.
+             */
+            animeNamingType: {
+                type: Number,
+                default: 0
+            },
+            /**
+             * Provide the custom naming type. -Like sports, anime, air by date- description.
+             * If none provided we asume this is the default episode naming component.
+             * And that means there will be no checkbox available to enable/disable it.
+             */
+            type: {
+                type: String,
+                default: ''
+            },
+            /**
+             * Used icw with the type property.
+             * If a type has been passed, the `enabled` property can be used to toggle the visibilty of the name-pattern settings. 
+            */
+            enabled: {
+                type: Boolean,
+                default: true
+            }
+        },
+        data() {
+            return {
+                presets: [],
+                availableMultiEpStyles: [],
+                pattern: '',
+                customName: '',
+                showLegend: false,
+                namingExample: '',
+                namingExampleMulti: '',
+                isEnabled: false,
+                isMulti: false,
+                selectedMultiEpStyle: 1,
+                animeType: 0
+            }
+        },
+        methods: {
+            getDateFormat(format) {
+                return dateFns.format(new Date(), format);
+            },
+            async testNaming(pattern, selectedMultiEpStyle, animeType) {
+                console.debug('Test pattern ' + pattern + ' for ' + (selectedMultiEpStyle) ? 'multi' : 'single' + ' ep');
+                const params = {
+                    pattern: pattern,
+                    anime_type: animeType
+                }
+
+                if (selectedMultiEpStyle) {
+                    params.multi = selectedMultiEpStyle;
+                }
+
+                try {
+                    const result = await apiRoute.get('config/postProcessing/testNaming', {params: params});
```

Comment:
If you switch this to `return apiRoute.get('config/postProcessing/testNaming', {params: params}).then(res => res.data);` then you wouldn't need the `async`.

Reply:
Also since this is async you should return something yet if this throws the catch is just going to log.

Reply:
What should I return here then?